docs/designs/allreduce.md

```diff
@@ -1,14 +1,215 @@
-# Design for Allreduce Support
+# Design for Allreduce-based Training Support
 
-This document describes the design for supporting Allreduce in ElasticDL. Note that this is still a work-in-progress.
+This document describes the design for supporting allreduce-based distributed training in ElasticDL.
+
+## Motivation
+
+While distributed training based on parameter servers can support training very large models and datasets by adding
+more workers and parameter servers as described in [parameter server design doc](parameter_server.md), there are additional
+challenges involved in order to optimize the performance, for example:
+
+* It is not easy to identify the right ratio of the number of workers to the number of parameter servers. For example,
```

Comment:
What's more, how to set the memory quota of worker and parameter server is a fine tuning work. We need avoid OOM as well as wasting memories.

Reply:
Thanks. Added!